pylearn2/datasets/hdf5.py

```diff
@@ -2,338 +2,467 @@
+            'No dataset was loaded. Please make sure that sources is a list '
+            'with at least one value and that the provided values are keys of '
+            'the dataset you are trying to load.')
+        super(HDF5Dataset, self).__init__(**kwargs)
 
-    def get_dataset(self, dataset, load_all=False):
+    def _read_hdf5(self, sources, aliases, load_all=False, use_h5py=True):
         """
-        Get a handle for an HDF5 dataset, or load the entire dataset into
-        memory.
+        Loads elements from an HDF5 dataset using either h5py or tables. It can
+        load either the whole object in memory or a reference to the object on
+        disk, depending on the load_all parameter. Returns a list of objects.
 
         Parameters
         ----------
-        dataset : str
-            Name or path of HDF5 dataset.
+        sources : list of str
+            List of HDF5 keys corresponding to the data to be loaded.
         load_all : bool, optional (default False)
             If true, load dataset into memory.
+        use_h5py: bool, optional (default True)
+            If true uses h5py, else tables.
         """
-        if load_all:
-            data = self._file[dataset][:]
+        data = alias_dict()
+        if use_h5py:
+            for s, a in safe_zip(sources, aliases):
+                if load_all:
+                    data[s, a] = self._fhandler[s][:]
+                else:
+                    data[s, a] = self._fhandler[s]
+                    # hdf5 handle has no ndim
+                    data[s].ndim = len(data[s].shape)
         else:
-            data = self._file[dataset]
-            data.ndim = len(data.shape)  # hdf5 handle has no ndim
+            for s, a in safe_zip(sources, aliases):
+                if load_all:
+                    data[s, a](self._fhandler.getNode('/', s)[:])
+                else:
+                    data[s, a] = self._fhandler.getNode('/', s)
         return data
 
-    def iterator(self, *args, **kwargs):
+    @wraps(Dataset.iterator, assigned=(), updated=())
+    def iterator(self, mode=None, data_specs=None, batch_size=None,
+                 num_batches=None, rng=None, return_tuple=False, **kwargs):
         """
-        Get an iterator for this dataset.
-
-        The FiniteDatasetIterator uses indexing that is not supported by
-        HDF5 datasets, so we change the class to HDF5DatasetIterator to
-        override the iterator.next method used in dataset iteration.
-
-        Parameters
-        ----------
-        WRITEME
+        data_specs : tuple, optional
+            A `(space, source)` tuple. See :ref:`data_specs` for a full
+            description. If set to None, the aliases (or sources) and spaces
+            provided when the dataset object has been created will be used.
+        return_tuple : bool, optional
+            Always return a tuple, even if there is exactly one source
+            of data being returned. Defaults to `False`.
```

Comment:
Yes, I specified just the parameter that are not present in the inherited method.